packages/malabi/src/remote-runner-integration/fetch-remote-telemetry.ts

```diff
@@ -11,16 +14,26 @@ interface FetchRemoteTelemetryProps {
  * @param fetchRemoteTelemetryProps Props for fetching remote telemetry
  * @param fetchRemoteTelemetryProps.portOrBaseUrl port number, or entire base url, where the endpoint is hosted at.
  */
-const fetchRemoteTelemetry = async ({ portOrBaseUrl } : FetchRemoteTelemetryProps): Promise<TelemetryRepository> => {
+const fetchRemoteTelemetry = async ({ portOrBaseUrl, currentTestTraceID } : FetchRemoteTelemetryProps): Promise<TelemetryRepository> => {
     try {
-        const baseUrl = typeof portOrBaseUrl === 'string' ? portOrBaseUrl : `http://localhost:${portOrBaseUrl}`;        
-        const res = await require('axios').get(`${baseUrl}/malabi/spans`, {
-            transformResponse: (res) => {
+        const baseUrl = typeof portOrBaseUrl === 'string' ? portOrBaseUrl : `http://localhost:${portOrBaseUrl}`;
+        const axios = require('axios');
+        const res = await axios.get(`${baseUrl}/malabi/spans`, {
+            transformResponse: (res: any) => {
                 return res;
             },
         });
-        const protoFormatted = collectorTraceV1Transform.fromJsonEncodedProtobufFormat(res.data);
-        const spans = collectorTraceV1Transform.fromProtoExportTraceServiceRequest(protoFormatted);
+
+        let spans;
+
+        if (process.env.MALABI_STORAGE_BACKEND === StorageBackend.Jaeger) {
+            const spansInJaegerFormat = JSON.parse(res.data).filter(({ traceID }) => traceID === currentTestTraceID)[0].spans;
+            spans = spansInJaegerFormat.map(jaegerSpan => convertJaegerSpanToOtelReadableSpan(jaegerSpan));
```

Comment:
consider always returning otel data from service. then the test runner is agnostic to the storage backend